Allow public-key-only entries in keyring

Quexington opened the pull request:
This adds support for entries in the keyring that have no secret information associated with them.  This is perhaps a silly choice to keep in the keyring since it is public information but since the keyring is the standard area where we manage keys right now it seems like the logical choice at least in the short term.

Quexington committed:
Allow public-key-only entries in keyring

Quexington committed:
Merge branch 'quex.remove_signing_from_wallet' into quex.allow_only_pks

Quexington committed:
Merge branch 'quex.remove_signing_from_wallet' into quex.allow_only_pks

Quexington committed:
fix test coverage

Quexington committed:
Merge branch 'quex.remove_signing_from_wallet' into quex.allow_only_pks

Quexington committed:
Merge branch 'quex.remove_signing_from_wallet' into quex.allow_only_pks

Quexington committed:
Merge branch 'quex.remove_signing_from_wallet' into quex.allow_only_pks

Quexington committed:
Merge branch 'quex.remove_signing_from_wallet' into quex.allow_only_pks

Quexington committed:
Merge branch 'quex.remove_signing_from_wallet' into quex.allow_only_pks

emlowe commented on chia/daemon/keychain_proxy.py:
```diff
@@ -348,6 +373,42 @@ async def get_key_for_fingerprint(self, fingerprint: Optional[int]) -> Optional[
+    async def get_public_key_for_fingerprint(self, fingerprint: Optional[int]) -> Optional[G1Element]:
+        """
+        Locates and returns a private key matching the provided fingerprint
+        """
```
Typo in docstring - this returns a public key

emlowe commented on chia/daemon/keychain_server.py:
```diff
@@ -234,6 +240,37 @@ async def add_private_key(self, request: Dict[str, Any]) -> Dict[str, Any]:
+            return {
+                "success": False,
+                "error": KEYCHAIN_ERR_KEYERROR,
+                "error_details": {"message": f"The word '{e.args[0]}' is incorrect.'", "word": e.args[0]},
```
this seems a curious error message - what `word` are you referring to here?

Quexington replied:
Not sure but I didn't write it and (now) didn't touch it so none of my business.

emlowe commented on chia/util/keychain.py:
```diff
@@ -362,6 +366,36 @@ def add_private_key(self, mnemonic: str, label: Optional[str] = None) -> Private
+                self.service,
+                get_private_key_user(self.user, index),
+                bytes(key).hex(),
+            )
```
I must admit to being a little confused what set_passphrase is doing here, but I note its used in add_private_key as well where it is equally confusing

emlowe replied:
Ok, this is oldish code to maintain API consistency with the keyring python module. Could be refactored out since we don't put "keys as a passphrase" anymore into the OS keyring...

emlowe commented on chia/daemon/keychain_proxy.py:
```diff
@@ -348,6 +373,42 @@ async def get_key_for_fingerprint(self, fingerprint: Optional[int]) -> Optional[
+    async def get_public_key_for_fingerprint(self, fingerprint: Optional[int]) -> Optional[G1Element]:
+        """
+        Locates and returns a private key matching the provided fingerprint
+        """
```
docstring typo - "returns a public key"

emlowe replied:
dup

emlowe commented on chia/daemon/keychain_server.py:
```diff
@@ -234,6 +240,37 @@ async def add_private_key(self, request: Dict[str, Any]) -> Dict[str, Any]:
+            return {
+                "success": False,
+                "error": KEYCHAIN_ERR_KEYERROR,
+                "error_details": {"message": f"The word '{e.args[0]}' is incorrect.'", "word": e.args[0]},
```
what `word` is this referring to - seems a curious error message

emlowe reviewed (commented):
Can we do a little more to remove the `no cover` pragmas

dependabot[bot] committed:
build(deps): bump cryptography from 42.0.2 to 42.0.4 (#17596)

Bumps [cryptography](https://github.com/pyca/cryptography) from 42.0.2 to 42.0.4.
- [Changelog](https://github.com/pyca/cryptography/blob/main/CHANGELOG.rst)
- [Commits](https://github.com/pyca/cryptography/compare/42.0.2...42.0.4)

---
updated-dependencies:
- dependency-name: cryptography
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Quexington committed:
Consolidate add_private_key and add_public_key

Quexington committed:
Consolidate get_key_for_fingerprint and get_public_key_for_fingerprint

Quexington committed:
Test fix

Quexington committed:
Merge remote-tracking branch 'origin/main' into catchup/long_lived_vault_from_main_368fecc17eb902551d21fad4e39b0f26befbbe67

Quexington committed:
Merge branch 'catchup/long_lived_vault_from_main_368fecc17eb902551d21fad4e39b0f26befbbe67' into quex.allow_only_pks
